dspace-spring-rest/src/main/java/org/dspace/app/rest/RelationshipTypeRestController.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.app.rest.converter.RelationshipTypeConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.RelationshipTypeRest;
+import org.dspace.app.rest.model.RelationshipTypeRestWrapper;
+import org.dspace.app.rest.model.hateoas.RelationshipTypeResourceWrapper;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.EntityType;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller will handle all the incoming calls on the api/core/entitytypes/{id}/relationshiptypes endpoint
+ * where the id parameter can be filled in to match a specific entityType and then get all the relationshipTypes
+ * for the given EntityType
+ */
+@RestController
+@RequestMapping("/api/core/entitytypes/{id}/relationshiptypes")
+public class RelationshipTypeRestController {
```

Comment:
This has been included in https://jira.duraspace.org/browse/DS-4225

Reply:
@abollini : Do we have examples elsewhere of the "usual way" that you are talking about? Just like with your other comment, I was looking around to see how this has been implemented on other endpoints (to see how hard this would be to refactor).  But, I was not able to find a good example.

Reply:
https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java#L749

it is based on the structure of the REST resource and/or the LinkMethod defined on the repository class